### PR TITLE
Dynamic axis barchart added

### PR DESCRIPTION
**Pull request by hjkmarshall**
Barchart component small refactor to be reusable
New chart type added

### CODE REVIEW

File: src/components/vanilla/charts/DynamicAxisBar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  const xAxis = xAxisOptions.find(item => value === item.title);
+  const updatedProps = {...props, xAxis: xAxis };
+
+  return (
+    <Container
+      {...props}
+      className="overflow-y-hidden">
+      <div className="flex h-[60px] w-full">
+          <div>
+            <select className="relative px-3 rounded-xl w-full min-w-[50px] h-10 border border-[#DADCE1] flex items-center" value={value} onChange={(e) => handleChange(e.target.value)} style={{ position: 'relative', height: '40px', width: 'auto'}}>
```
Comment (kuka-radovan) [inline src/components/vanilla/charts/DynamicAxisBar/index.tsx:34]: Is that style definition necessary? position is duplicated with class `relative`, height could be replaced by class `h-[50px]` a width with `w-auto`

File: src/components/vanilla/charts/BarChart/components/BarChart.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  Tooltip
+} from 'chart.js';
+import ChartDataLabels from 'chartjs-plugin-datalabels';
+import React from 'react';
+import { Bar } from 'react-chartjs-2';
+
+import { COLORS, EMB_FONT, LIGHT_FONT, SMALL_FONT_SIZE, DATE_DISPLAY_FORMATS } from '../../../../constants';
+import formatValue from '../../../../util/format';
+import getBarChartOptions from '../../../../util/getBarChartOptions';
+import Container from '../../../Container';
```
Comment (kuka-radovan) [inline src/components/vanilla/charts/BarChart/components/BarChart.tsx:21]: It looks like this is not used anywhere in the code

File: src/components/vanilla/charts/BarChart/index.tsx [hunk 3]
```diff
@@ -1,92 +1,17 @@
...
-type Props = {
-  results: DataResponse;
-  title: string;
-  xAxis: Dimension;
-  metrics: { name: string; title: string }[];
-  granularity: string;
-};
+import BarChart from './components/BarChart';
 
 export default (props: Props) => {
```
Comment (kuka-radovan) [inline src/components/vanilla/charts/BarChart/index.tsx:4]: Props type is not defined here.

File: src/components/vanilla/charts/BarChart/components/BarChart.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  return (
+      <Bar
+        height="100%"
+        options={getBarChartOptions({ ...props, stacked: false })}
+        data={chartData(props)}
+      />
+  );
+}
+
+function chartData(props: Props, value): ChartData<'bar'> {
```
Comment (kuka-radovan) [inline src/components/vanilla/charts/BarChart/components/BarChart.tsx:58]: `value` is not used anywhere in this code

File: src/components/vanilla/charts/BarChart/components/BarChart.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  const { results, xAxis, metrics, granularity, xAxisOptions } = props;
```
Comment (kuka-radovan) [inline src/components/vanilla/charts/BarChart/components/BarChart.tsx:59]: `xAxisOptions` is not used and even props doesn not know this key.

File: src/components/vanilla/charts/DynamicAxisBar/index.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import { useEmbeddableState } from '@embeddable.com/react';
+import React, { useState, useEffect, useRef } from 'react';
```
Comment (kuka-radovan) [inline src/components/vanilla/charts/DynamicAxisBar/index.tsx:2]: `useEffect` and  `useRef` are not used in the code

File: src/components/vanilla/charts/DynamicAxisBar/index.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import Container from '../../Container';
+import Dropdown from '../../controls/Dropdown';
```
Comment (kuka-radovan) [inline src/components/vanilla/charts/DynamicAxisBar/index.tsx:4]: `Dropdown` is not used in the code

File: src/components/vanilla/charts/DynamicAxisBar/index.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+import BarChart from '../BarChart/components/BarChart';
+
+export default (props: Props) => {
```
Comment (kuka-radovan) [inline src/components/vanilla/charts/DynamicAxisBar/index.tsx:7]: `Props` type is not defined here

File: src/components/vanilla/charts/DynamicAxisBar/DynamicAxisBar.emb.ts [hunk 9]
```diff
@@ -0,0 +1,160 @@
+import { EmbeddedComponentMeta, Inputs, defineComponent } from '@embeddable.com/react';
+import { loadData, isMeasure, isDimension, Dimension, Measure } from '@embeddable.com/core';
+
+import Component, { Props } from './index';
```
Comment (kuka-radovan) [inline src/components/vanilla/charts/DynamicAxisBar/DynamicAxisBar.emb.ts:4]: `./index` does not export Props. It even does not define Props type.

File: src/components/vanilla/charts/DynamicAxisBar/DynamicAxisBar.emb.ts [hunk 10]
```diff
@@ -0,0 +1,160 @@
+import { EmbeddedComponentMeta, Inputs, defineComponent } from '@embeddable.com/react';
+import { loadData, isMeasure, isDimension, Dimension, Measure } from '@embeddable.com/core';
```
Comment (kuka-radovan) [inline src/components/vanilla/charts/DynamicAxisBar/DynamicAxisBar.emb.ts:2]: `isMeasure, isDimension, Dimension, Measure` are not used

File: src/components/vanilla/charts/DynamicAxisBar/DynamicAxisBar.emb.ts [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+                    operator: 'notEquals',
+                    value: [null]
+                }],
+                orderBy: [
+                    { 
+                        property: selectedDimension, 
+                        direction: 'desc' 
+                    }
+                ],
+                limit: inputs.limit || 50
```
Comment (kuka-radovan) [inline src/components/vanilla/charts/DynamicAxisBar/DynamicAxisBar.emb.ts:144]: limit is not among inputs

File: src/components/vanilla/charts/DynamicAxisBar/index.tsx (hunk 1 above)
Comment (hjkmarshall) [reply src/components/vanilla/charts/DynamicAxisBar/index.tsx:34]: @kuka-radovan good point, but I added this due to a really strange bug with stencil + lastpass (that our team know about). Without this, the layout doesn't work properly when you click for the first time.

Comment (hjkmarshall) [reply src/components/vanilla/charts/DynamicAxisBar/index.tsx:34]: But I will remove this anyway, as it's such an edge case.

File: src/components/vanilla/charts/BarChart/components/BarChart.tsx (hunk 2 above)
Comment (hjkmarshall) [reply src/components/vanilla/charts/BarChart/components/BarChart.tsx:21]: Removed

File: src/components/vanilla/charts/BarChart/index.tsx (hunk 3 above)
Comment (hjkmarshall) [reply src/components/vanilla/charts/BarChart/index.tsx:4]: Updated

File: src/components/vanilla/charts/BarChart/components/BarChart.tsx (hunk 4 above)
Comment (hjkmarshall) [reply src/components/vanilla/charts/BarChart/components/BarChart.tsx:58]: Removed

File: src/components/vanilla/charts/BarChart/components/BarChart.tsx (hunk 5 above)
Comment (hjkmarshall) [reply src/components/vanilla/charts/BarChart/components/BarChart.tsx:59]: Removed

File: src/components/vanilla/charts/DynamicAxisBar/index.tsx (hunk 8 above)
Comment (hjkmarshall) [reply src/components/vanilla/charts/DynamicAxisBar/index.tsx:7]: Done

File: src/components/vanilla/charts/DynamicAxisBar/DynamicAxisBar.emb.ts (hunk 9 above)
Comment (hjkmarshall) [reply src/components/vanilla/charts/DynamicAxisBar/DynamicAxisBar.emb.ts:4]: Removed

File: src/components/vanilla/charts/DynamicAxisBar/DynamicAxisBar.emb.ts (hunk 11 above)
Comment (hjkmarshall) [reply src/components/vanilla/charts/DynamicAxisBar/DynamicAxisBar.emb.ts:144]: Removed

File: src/components/vanilla/charts/DynamicAxisBar/index.tsx (hunk 1 above)
Comment (kuka-radovan) [reply src/components/vanilla/charts/DynamicAxisBar/index.tsx]: I have no problem with that if it is intentional. I don't know the circumstances.

File: src/components/vanilla/charts/DynamicAxisBar/index.tsx (hunk 6 above)
Comment (hjkmarshall) [reply src/components/vanilla/charts/DynamicAxisBar/index.tsx]: @kuka-radovan I removed useRef - useEffect is used.

**Review by kuka-radovan — APPROVED** [review]

**Pull request merged**